### PR TITLE
[planning_interface] Unit Test for ByteString class

### PR DESCRIPTION
**Pull request by gleichdick**
### Description

I just found an unit test for the ByteString class in my workspace, dunno whether it makes sense to include it upstream...

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/planning_interface/test/test_serialize_msg_python.cpp [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+    if (!py_data)
+      return false;
+    Py_ssize_t size = PyBytes_GET_SIZE(obj);
+    if (size != data.size())
+      return false;
+    return std::memcmp(py_data, &data[0], size) == 0;
+  }
+  bool compare_bytes(const ByteString& s)
+  {
+    const std::vector<unsigned char> testdata{ 0xff, 0xef, 0x00, 0x10 };
```
Comment (rhaschke) [inline moveit_ros/planning_interface/test/test_serialize_msg_python.cpp]: What is actually test here? An arbitrary? Bytestring against a fixed uchar vector?

**Comment by gleichdick** [comment]
These Testcases where made to ensure that the message passing between C++ and Python didn't break. That's why I had to embed the Python Interpreter and to exchange data with it.
I added some comments to give a hint about the test cases, but I don't mind not merging this PR tbh.

File: moveit_ros/planning_interface/test/test_serialize_msg_python.cpp (hunk 1 above)
Comment (gleichdick) [reply moveit_ros/planning_interface/test/test_serialize_msg_python.cpp]: this method is exposed to Python and called with `b'\\xff\\xef\\x00\\x10'` as argument from within Python. Then the `bytes` Python instance will be converted to C++ `py_bindings_tools::ByteString` by boost, the converted Object is then tested.
So we know that embedded NULL's won't be dropped.

**Comment by gleichdick** [comment]
Rebased onto master

**Comment by felixvd** [comment]
If you restart Travis it should build now.

**Comment by rhaschke** [comment]
Closing and reopening to trigger Travis.

**Review by rhaschke — APPROVED** [review]
Just noticed that this PR is still pending. What a shame!
Generally, I approve. However, designing the test a C++ based test only, makes it rather complicated. Splitting it into a C++ part (providing `ByteStringTestHelper`) and a python part (actually performing the tests) would, I think, considerably simplify the code.
If you agree, @gleichdick, and are (still) willing to work on this, I really would appreciate that.
If not, I'm fine to merge this as is. I will wait another few days for your response.

**Comment by gleichdick** [comment]
Thank you for your comments, I will try to do my best. I'm likely to be
quarantined because of Covid-19, so I'll have plenty of time 😅

Robert Haschke <notifications@github.com> schrieb am Fr., 9. Okt. 2020,
10:45:

**Comment by rhaschke** [comment]
Great thanks. I hope you are fine regarding Covid-19!

**Comment by gleichdick** [comment]
Thanks. First Covid-Test was negative :relaxed: 
Splitting the test in Python and C++ components was a very good hint, thank you very much. #2362 should do it.

**Pull request closed without merging**